pkg/policymappings/policymappings.go

```diff
@@ -0,0 +1,145 @@
+package policymappings
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"reflect"
+
+	"github.com/fairwindsops/insights-cli/pkg/directory"
+	cliversion "github.com/fairwindsops/insights-cli/pkg/version"
+	"github.com/imroc/req"
+	"github.com/samber/lo"
+	"github.com/sirupsen/logrus"
+	"github.com/xlab/treeprint"
+	"gopkg.in/yaml.v2"
+)
+
+// BuildPolicyMappingsTree builds a tree for policy-mapping
+func BuildPolicyMappingsTree(policyMappings []PolicyMapping) (treeprint.Tree, error) {
+	tree := treeprint.New()
+	policyMappingsBranch := tree.AddBranch("policy-mappings")
+	for _, policyMapping := range policyMappings {
+		enabled := "enabled"
+		if policyMapping.Spec.Enabled != nil && !*policyMapping.Spec.Enabled {
+			enabled = "disabled"
+		}
+		policyMappingsBranch.AddBranch(fmt.Sprintf("%s (%s)", policyMapping.Name, enabled))
+	}
+	return tree, nil
+}
+
+// PushPolicyMappings pushes policy-mapping to insights
+func PushPolicyMappings(pushDir, org, insightsToken, host string, deleteMissing, dryRun bool) error {
+	logrus.Debugln("Pushing policy-mapping")
+	_, err := os.Stat(pushDir)
+	if err != nil {
+		return err
+	}
+
+	existingPolicyMappings, err := FetchPolicyMappings(org, insightsToken, host)
+	if err != nil {
+		return fmt.Errorf("error during API call: %w", err)
+	}
+
+	upserts, deletes, err := comparePolicyMappings(pushDir, existingPolicyMappings)
+	if err != nil {
+		return fmt.Errorf("unable to compare and push policy-mapping to Insights: %w", err)
+	}
+
+	for _, policyMapping := range upserts {
+		logrus.Infof("upsert policy-mapping: %s", policyMapping.Name)
+		if !dryRun {
+			err = upsertPolicyMapping(org, insightsToken, host, policyMapping)
```

Comment:
Wondering if we could have a way to do this in a batch?

Reply:
There are no APIs for it just yet, frontend will use single API's, the payload is lightweight and orgs shouldn't have more than a dozen but could be an improvement

Reply:
alright, thought they would have a lot of them